### PR TITLE
test(e2e): isDeadlineExceeded

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3380

## Testing

I make it past the place where I hit the error in the issue. Now I'm hitting a different error which makes me think this PR resolved at least one issue.

```
test-e2e2024/04/24 15:42:27 Starting testnets
test-e2e2024/04/24 15:45:31 Running txsim
test-e2e2024/04/24 15:46:01 Reading blockchain
test-e2e2024/04/24 15:46:51 --- ERROR E2ESimple: expected at least 10 transactions, got 1
exit status 1
make: *** [test-e2e] Error 1
```

### CODE REVIEW

File: test/e2e/simple.go
Comment (rach-id) [inline test/e2e/simple.go:54]: context deadline exceeded is only one type of errors that txsim can return, and there are others. Why are we expecting it to be the only one?

**Review by rach-id — COMMENTED** [review]
> I make it past the place where I hit the error in the issue. Now I'm hitting a different error which makes me think this PR resolved at least one issue.

I personally prefer to leave this PR open until all the issues are resolved, but I defer to your judgement

File: test/e2e/simple.go
Comment (rach-id) [reply test/e2e/simple.go:54]: if that's the happy path, why would we need to execute the rest of the tests? 

also, why wouldn't it be something like:

```

if err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...); !isDeadlineExceeded(err) {
	return fmt.Errorf("expected error to contain context deadline exceeded but got %w", err)
}

=>

err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
assert.Falsef(t, isDeadlineExceeded(err), "expected error to contain context deadline exceeded but got %w", err)
```
or similar?

File: test/e2e/simple.go
Comment (cmwaters) [inline test/e2e/simple.go:55]: Does this not work because the error gets wrapped? If that is the case I would prefer that the error gets unwrapped / or otherwise handled in txsim itself

**Review by cmwaters — CHANGES_REQUESTED** [review]

File: test/e2e/simple.go
Comment (rootulp) [reply test/e2e/simple.go:55]: I think a context deadline exceeded error gets wrapped in an RPC error but TBH then I would expect the original implementation `errors.Is` to have worked.

I can hold off on merging this b/c I don't have a convincing way to prove this helped or hurt b/c the test is flaky for other reasons outside this issue.

File: test/e2e/simple.go
Comment (rootulp) [reply test/e2e/simple.go:54]: oh I kinda see what you mean. I think you meant:

```go
err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
assert.True(t, isDeadlineExceeded(err), "expected error to contain context deadline exceeded but got %w", err)
```

but yea that would make it clearer that the test expects a context deadline exceeded error.

File: test/e2e/simple.go
Comment (cmwaters) [reply test/e2e/simple.go:55]: Yeah I think we should then unwrap it in txsim and just return the raw deadline exceeded error

**Comment by rootulp** [comment]
Not planning on working on this

**Pull request closed without merging**
